README.md

```diff
@@ -20,11 +20,42 @@ Using the stock OSX version of git will likely cause problems. You probably want
     homesick symlink  dotfiles
 
 ### Optional
-If you're going to use hitch install it by running bundle in the dotfiles directory
+
+If you're going to use hitch or other ruby scripts in our dotfiles, install gem
+dependencies by running bundle in the dotfiles directory
 
     cd ~/.homesick/repos/dotfiles/home
     bundle
 
+### Git Submodules
+
+In order for vim plugins (and a couple tmux ones) to work correctly, you have
+to go through a few steps.
+
+Enter your git user details into `~/.gitconfig.d/user` and make sure you have
+an ssh key added to Github.
+
+```
+[user]
+  name = [YOUR NAME HERE]
+  email = [YOUR EMAIL HERE]
+```
+
+Run `git submodule update` from the `~/.homesick/repos/dotfiles/home/.vim/bundle`
+directory.
+
+If at this point you have a bunch of unstaged changes in the gitsubmodule
+directories, this is a known quirk with submodules and can be fixed. Until you
+do, opening vim will likely result in a bunch of errors.
+
+Note: These steps assume that the unstaged changes are a bunch of deletions in
+each submodule directory. If this is not the case for you, then modify the
+`foreach` commands to execute the correct git commands to undo the changes you
+have.
+
+1. Run `git submodule foreach --recursive git reset`
+2. Run `git submodule foreach --recursive git co .`
```

Comment:
If you scroll to the last section of [this article](https://longair.net/blog/2010/06/02/git-submodules-explained/), "Modified submodules in 1.7.0 and later", it shows exactly what happened to me. Basically, even though the HEAD SHA is no different, git marks each submodule as dirty, which left a lot of noise in my diff. Definitely wouldn't want that committed to the project, and I don't think it is committed when you're just git adding your own files outside of the submodule directories, but it's annoying to see locally. So my thought was to provide steps to try to clear that out for people so it's not annoying. Can definitely add more explanation in the README, or maybe I'm totally mistaken on all of this but this is what happened to me.

Here is the [Git commit](https://github.com/git/git/commit/ee6fc514f2df821c2719cc49499a56ef2fb136b0) in question that I think caused this, which is mentioned by the above article. And the [Stack Overflow post](http://stackoverflow.com/questions/4873980/git-diff-says-subproject-is-dirty/4874323) where I found the article and which provided insight into how I could fix the problem.